src/components/Facility/ConsultationDoctorNotes/index.tsx

```diff
@@ -54,40 +55,52 @@ const ConsultationDoctorNotes = (props: ConsultationDoctorNotesProps) => {
 
   const initialData: PatientNoteStateType = {
     notes: [],
-    cPage: 1,
-    totalPages: 1,
     facilityId: facilityId,
     patientId: patientId,
   };
   const [state, setState] = useState(initialData);
+  const queryClient = useQueryClient();
+
+  const addNoteMutation = useMutation({
+    mutationFn: async (noteField: string) => {
+      const { res, data } = await request(routes.addPatientNote, {
+        pathParams: { patientId },
+        body: {
+          note: noteField,
+          thread,
+          consultation: consultationId,
+          reply_to: reply_to?.id,
+        },
+      });
+      if (res?.status === 201 && data) return data;
+      throw new Error("Failed to add note");
+    },
+    onSuccess: (newNote) => {
+      setState((prevState) => ({
+        ...prevState,
+        notes: [newNote, ...prevState.notes],
+      }));
+      setReplyTo(undefined);
+      Notification.Success({ msg: "Note added successfully" });
+
+      queryClient.invalidateQueries({
+        queryKey: ["notes"],
+      });
+    },
+    onError: () => {
+      Notification.Error({ msg: "Failed to add note" });
+    },
```

Comment:
error messages need not be handled. We have a global handler for that, which describes the exact reason why it failed if detectable instead.